src/odemis/acq/move.py

```diff
@@ -834,160 +1001,410 @@ def _doCryoSwitchSamplePosition(future, target):
-    Perform the stage reference procedure
-    :param future: cancellable future of the reference procedure
-    :param component: Either the stage or the align component
-    :raises CancelledError: if the reference is cancelled
-    """
-    try:
-        with future._task_lock:
-            if future._task_state == CANCELLED:
-                logging.info("Reference procedure is cancelled.")
-                raise CancelledError()
-            logging.debug("Performing stage referencing.")
-            future._running_subf = component.reference(set(component.axes.keys()))
-        future._running_subf.result()
-    except Exception as error:
-        logging.exception(error)
-    if future._task_state == CANCELLED:
-        logging.info("Reference procedure is cancelled.")
-        raise CancelledError()
-
-
-def run_sub_move(future, component, sub_move):
-    """
-    Perform the sub moveAbs using the given component and axis->pos dict
-    :param future: cancellable future of the whole move
-    :param component: Either the stage or the align component
-    :param sub_move: the sub_move axis->pos dict
-    :raises TimeoutError: if the sub move timed out
-    :raises CancelledError: if the sub move is cancelled
-    """
-    try:
-        with future._task_lock:
-            if future._task_state == CANCELLED:
-                logging.info("Move procedure is cancelled before moving %s -> %s", component.name, sub_move)
-                raise CancelledError()
-
-            logging.debug("Performing sub move %s -> %s", component.name, sub_move)
-            future._running_subf = component.moveAbs(sub_move)
-        future._running_subf.result(timeout=MAX_SUBMOVE_DURATION)
-    except TimeoutError:
-        future._running_subf.cancel()
-        logging.exception("Timed out while moving %s -> %s", component.name, sub_move)
-        raise
-
-    if future._task_state == CANCELLED:
-        logging.info("Move procedure is cancelled after moving %s -> %s", component.name, sub_move)
-        raise CancelledError()
+        except CancelledError:
+            logging.info("CryoSwitchSamplePosition cancelled.")
+        except Exception:
+            logging.exception("Failure to move to {} position.".format(target_name))
+            raise
+        finally:
+            with future._task_lock:
+                if future._task_state == CANCELLED:
+                    raise CancelledError()
+                future._task_state = FINISHED
+
+
+class EnzelPostureManager(MicroscopePostureManager):
+    def __init__(self, microscope):
+        # Load components
+        self.stage = model.getComponent(role='stage')
+        self.align = model.getComponent(role='align')
+        # set linear axes and rotational axes used
+        self.axes = self.stage.axes
+        self.linear_axes = set(key for key in self.axes.keys() if key in {'x', 'y', 'z', 'm'})
+        self.rotational_axes = set(key for key in self.axes.keys() if key in {'rx', 'ry', 'rz', 'rm'})
+
```

Comment:
Need to check the metadata too. See the gui.cont.tabs for the (minimal) list of metadata required.

Reply:
Yes, please add this more complete test to the PostureManager.